sdk/storage/azblob/service/client_test.go

```diff
@@ -314,6 +316,41 @@ func (s *ServiceRecordedTestsSuite) TestSetPropertiesSetCORSMultiple() {
 	_require.NoError(err)
 }
 
+type userAgentTest struct{}
+
+func (u userAgentTest) Do(req *policy.Request) (*http.Response, error) {
+	const userAgentHeader = "User-Agent"
+
+	currentUserAgentHeader := map[string][]string(req.Raw().Header)[userAgentHeader]
```

Comment:
You can use `req.Raw().Header.Get()` method here. Something like this, https://github.com/Azure/azure-storage-fuse/blob/feature/track2Migration/component/azstorage/utils.go#L380-L382